analysis/quantifier-comparisons/08-genes-detected-comparison.Rmd

```diff
@@ -0,0 +1,405 @@
+  purrr::map_df(
+    ~ purrr::map_df(.x, scpcaTools::rowdata_to_df, .id = "quant_dir"),
+    .id = "alevin_resolution"
+  ) %>%
+  dplyr::left_join(quant_info)
+```
+
+## Differences in Gene detection between Alevin-fry and Cellranger
+
+### Overlap of Genes 
+
+As another comparison, let's look at the actual genes that are identified in each of the tools. 
+Let's see how many are lost when we use alevin-fry over cellranger and what types of genes they are. 
+
+```{r}
+# make sure rowdata only has genes that are found in more than 0% cells and > 0% mean gene expression
+rowdata_filtered <- rowdata_filtered %>%
+  dplyr::filter(detected > 0 & mean > 0)
+```
+
+
+```{r}
+# pull out genes that are detected and expressed in each of the tools
+cellranger_genes <- rowdata_filtered %>%
+  dplyr::filter(tool == "cellranger" & detected > 0 & mean > 0) %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+
+cr_like_genes <- rowdata_filtered %>%
+  dplyr::filter(alevin_resolution == "cr-like" & detected > 0 & mean > 0) %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+
+cr_em_genes <- rowdata_filtered %>%
+  dplyr::filter(alevin_resolution == "cr-like-em" & detected > 0 & mean > 0) %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+```
+
+
+```{r}
+# get intersection gene lists 
+# look at intersection between cellranger and alevin-fry cr-like
+cellranger_cr_intersect <- intersect(cellranger_genes, cr_like_genes)
+
+# look at intersection between cellranger and alevin-fry cr-like-em
+cellranger_cr_em_intersect <- intersect(cellranger_genes, cr_em_genes)
+
+# look at intersection between alevin-fry cr-like and alevin-fry cr-like-em
+cr_cr_em_intersect <- intersect(cr_like_genes, cr_em_genes)
+```
+
+```{r}
+# get unique gene lists by getting genes that overlap with all other tools first and then the set difference with the list of genes for that tool
+all_cellranger_overlap <- rowdata_filtered %>%
+  dplyr::filter(gene_id %in% cellranger_genes & tool =="alevin-fry") %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+cellranger_unique <- setdiff(cellranger_genes, all_cellranger_overlap)
+
+cr_overlap <- rowdata_filtered %>%
+  dplyr::filter(gene_id %in% cr_like_genes & alevin_resolution %in% c("cr-like-em", "not_alevin")) %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+cr_unique <- setdiff(cr_like_genes, cr_overlap)
+
+cr_like_overlap <- rowdata_filtered %>%
+  dplyr::filter(gene_id %in% cr_em_genes & alevin_resolution %in% c("cr-like", "not_alevin")) %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+cr_like_unique <- setdiff(cr_em_genes, cr_like_overlap)
+```
+
+
+```{r}
+# visualize overlap between gene sets based on tool
+rowdata_filtered[which(rowdata_filtered$alevin_resolution == "not_alevin"), "alevin_resolution"] <- "cellranger"
+
+gene_detect_df <- rowdata_filtered %>%
+  group_by(gene_id) %>%
+  dplyr::summarise(tools_detected = list(unique(alevin_resolution)))
+
+gene_detect_df %>%
+  dplyr::distinct(gene_id, .keep_all = TRUE) %>%
+  ggplot(aes(x = tools_detected)) +
```

Comment:
I know people do it all the time, but as a style thing, I hate piping into ggplot. 
As I was rewriting this, I think the `distinct()` call is not needed? The group_by/summarise should have taken care of that, right?

```suggestion
gene_detect_df <- rowdata_filtered %>%
  group_by(gene_id) %>%
  dplyr::summarise(tools_detected = list(unique(alevin_resolution))) 

ggplot(gene_detect_df, aes(x = tools_detected)) +
```